lumigator/jobs/inference/model_clients/huggingface_clients.py

```diff
@@ -13,7 +13,7 @@ class HuggingFaceModelClientFactory:
     """Factory class that creates the appropriate specialized client"""
 
     @staticmethod
-    def create(config: InferenceJobConfig) -> BaseModelClient:
+    def create(config: InferenceJobConfig, api_key: str | None = None) -> BaseModelClient:
```

Comment:
Is it necessary to have the `api_key: str | None = None` syntax everywhere, or could you have that in one place and then enforce that everyone else gets the api_key as `api_key: str | None`.

aka can we have the default constructor value be specified in a single location only?

Reply:
I get you, I suppose it's just because you don't have to use the static factory method to get the client, so I dunno if there's any harm in the default `None`. 🤔